### DISCUSSION TITLE
Shared Subscriptions with Durable Sessions EMQX 5.7

### DISCUSSION THREAD

**Discussion by ennc0d3** [emqx/emqx#13152] [Q&A] [answered]
I am running EMQX (CE v5.7.0) with durable_sessions.enable = true and durable_storage options set which works for regular persistent subscriptions. 

But, If i try the same for shared subscriptions (persistent session), EMQX rejects with (code: 158) Shared subscriptions not supported.  

* Is this a limitation? 
* Can we selectively disable durable_sessions just for shared subscriptions?

Thanks, Naga

**Comment by savonarola** [comment]
Hello!
The difference in implementation (durable vs. regular) is session-based rather than subscription-based.
That is, durability is on/off for clients, not for subscriptions.

So, to have access to regular shared subs, a client should connect with `ExpiryInterval` set to 0. Also, please note, if a client already has a not expired durable session, it will be reopened.

**Reply by ennc0d3** [reply]
Thanks,

> Hello! The difference in implementation (durable vs. regular) is session-based rather than subscription-based. That is, durability is on/off for clients, not for subscriptions.
>
I mean't the message persistence in EMQX, With EMQX 5.7.0 durable sessions/storage is available, if enabled the message are stored in Db, so the clients (with persistence sessions and subscription) get messages if they go offline and back again.

> So, to have access to regular shared subs, a client should connect with `ExpiryInterval` set to 0. Also, please note, if a client already has a not expired durable session, it will be reopened.
Yes, Thats is right

My concern is the clients(persistent session) can't subscribe to shared topics, with EMQX 5.7.0 when durable_session feature is enabled. 

When is it planned to be released?

**Reply by id** [reply]
Durable storage for shared subscriptions is a separate feature which is not built yet. So far, we do not have any planned release date.

**Comment by mikolajwalus** [comment]
Is there any update on durable sessions on shared subscription?

**Reply by zmstone** [reply]
Hi @mikolajwalus it will likely to be a part of 5.9 (release plan: March/April 2025)

**Reply by mikolajwalus** [reply]
That's fantastic! Is it planned to be Enterprise-only feature or it will be included in Open Source version too?

**Reply by mikolajwalus** [reply]
@zmstone just to remind about question.

Best regards

**Comment by ieQu1** [comment, marked as answer]
This feature has been added in 6.0.0.
